packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1446 @@
+  ///
+  /// Override this for custom thumb selection.
+  final RangeThumbSelector thumbSelector;
+
+  /// Lower bound on how close the thumbs can be together in dp.
+  ///
+  /// Use this only if you want to control the visual appearance of the thumbs
+  /// based on pixel value. To limit with the real values, the [values] should
+  /// be restricted by the parent
+  final double minThumbSeparation;
+
+  /// The callback used to create a semantic value from slider values.
+  ///
+  /// Defaults to formatting values as a percentage.
+  ///
+  /// This is used by accessibility frameworks like TalkBack on Android to
+  /// inform users what the currently selected value is with more context.
+  ///
+  /// {@tool sample}
+  ///
+  /// In the example below, a slider for currency values is configured to
+  /// announce a value with a currency label.
+  ///
+  /// ```dart
+  /// RangeSlider(
+  ///   value: _dollarsRange,
+  ///   min: 20.0,
+  ///   max: 330.0,
+  ///   onChanged: (RangeValues newValues) {
+  ///     setState(() {
+  ///       _dollarsRange = newValues;
+  ///     });
+  ///   },
+  ///   semanticFormatterCallback: (double newValue) {
+  ///     return '${_dollarsRange.start.round()} - ${_dollarsRange.end.round()} dollars';
+  ///   }
+  ///  )
+  /// ```
+  /// {@end-tool}
+  final RangeSemanticFormatterCallback semanticFormatterCallback;
+
+  /// Touch width for the tap boundary of the slider thumbs.
+  static const double minTouchTargetWidth = 48;
+
+  @override
+  _RangeSliderState createState() => _RangeSliderState();
+}
+
+class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin {
+  static const Duration enableAnimationDuration = Duration(milliseconds: 75);
+  static const Duration valueIndicatorAnimationDuration = Duration(milliseconds: 100);
+
+  // Animation controller that is run when the overlay (a.k.a radial reaction)
+  // changes visibility in response to user interaction.
+  AnimationController overlayController;
+
+  // Animation controller that is run when the value indicators change visibility.
+  AnimationController valueIndicatorController;
+
+  // Animation controller that is run when enabling/disabling the slider.
+  AnimationController enableController;
+
+  // Animation controllers that are run when transitioning between one value
+  // and the next on a discrete slider.
+  AnimationController startPositionController;
+  AnimationController endPositionController;
+  Timer interactionTimer;
+
+  @override
+  void initState() {
+    super.initState();
+    overlayController = AnimationController(
+      duration: kRadialReactionDuration,
+      vsync: this,
+    );
+    valueIndicatorController = AnimationController(
+      duration: valueIndicatorAnimationDuration,
+      vsync: this,
+    );
+    enableController = AnimationController(
+      duration: enableAnimationDuration,
+      vsync: this,
+    )..value = widget.onChanged != null ? 1.0 : 0.0;
```

Comment:
Can't this be done inline like the startPositionController?

```
    enableController = AnimationController(
      duration: enableAnimationDuration,
      vsync: this,
      value: widget.onChanged != null ? 1.0 : 0.0,
    );
```

Reply:
Done